### DISCUSSION TITLE
Is "load Collection" only needed to be executed once or needed to be called frequently (for any new stored data) to sync quickly?

### DISCUSSION THREAD

**Discussion by grimmerk** [milvus-io/milvus#28677] [Q&A and General discussion]
As title. I'm also unsure what is the current synchronization mechanism if I only call load collection once. It would be great to learn about this.

**Comment by yhmo** [comment]
Only call load() one time. It will automatically load the newly inserted data into memory.
Just make sure load() is after create_index(), and search() is after load().

All the following workflows are ok:
- create_collection ---> **create_index** ---> **load** ---> insert ---> insert ..... ---> search ---> insert ---> search
- create_collection ---> insert ---> ... insert --> **create_index** ---> **load** ---> insert ..... ---> search ---> insert ---> search
- create_collection ---> **create_index** ---> insert ---> insert ..... ---> **load** ---> search ---> insert ---> search

**Comment by 631068264** [comment]
@yhmo Hello I've encountered some problems recently  about `load()`.

For example , I have two process A,B  , they all `load` same collection  C in the beginning  for search.

A  ---> load ---> search ---> delete ---> search ( new data)

B  ---> load ---> search  ---> long time  after delete    ---> search( old data)


A can get new data, but B not . **For this case ，I can only  reload  C  in B.**   But it's  strange. Get different result when two process to operate the same  collection.


I can load before search every time to avoid this problem, but  load collection take a long time when stored data more than one million.

I don't know why user should load  collection manually in the client code, why not in server when it start ?

**Reply by xiaofan-luan** [reply]
you need to tune you consistency level if you need to see instant change.
By default we use bounded staleness and there will be a 3 seconds lag

**Reply by 631068264** [reply]
@xiaofan-luan  In my case,  I use langchain which default consistency_level is **Session**. 

- So I  should change it to Bounded staleness ? 
-  Can I change the consistency_level when the collection exist ?  And how ?
-  Or I only set consistency_level when create collection

**Reply by yhmo** [reply]
In langchain, you can pass consistency_level=XXXX to any search interface of VectorStore such as similarity_search(), to overwrite the default level, for example.
```
    milvus = Milvus.from_documents(
        documents=docs,
        embedding=HuggingFaceEmbeddings(model_name=model_name),
        collection_name="AAA",
        connection_args={"host": "localhost", "port": 19530}
    )

    docs = milvus.similarity_search(
        query="Ergonomic zero administration knowledge user",
        consistency_level="Strong",
        k=5)
```


The options of consistency_level are listed here: https://milvus.io/docs/consistency.md#Consistency-levels

To force seach the new data right after you insert/delete, the Strong level is the right choice, but with this level the search() interface needs to wait 200~400ms until the previous dml actions are consumed.
